flask_app/main.py

```diff
@@ -1,39 +1,23 @@
-from flask import Flask, escape, request, render_template
+from flask import Flask, jsonify, make_response
 import get_data
 
 app = Flask(__name__)
 
 
-@app.route("/")
-def index():
-    return render_template("index.html")
-
-
-@app.route("/zastepstwa")
-def zastepstwa():
-    teachers = get_data.main()
-    return render_template("z.html", teachers=teachers)
-
-
-@app.route("/zastepstwa-pelne")
-def zastepstwa_original():
+@app.route("/api/", methods=["GET"])
+def get_all():
     teachers = get_data.main()
-    return render_template("z-pelne.html", teachers=teachers, i=0, len=len)
-
+    return jsonify({'data': teachers})

```

Comment:
By list of teachers, I mean just the teacher's names, without the substitutions.
```suggestion
    return jsonify({'data': teachers.keys()})
```